File: Spark/python/word_count_rdd.py
from pyspark.conf import SparkConf
from pyspark.context import SparkContext

import re

sparkConf = SparkConf().master('local[*]') \
    .appName('WordCount_RDD_Python')
sc = SparkContext(sparkConf)

file_path = 'file://' + '/path/to/a/textfile'
rdd = sc.textFile(file_path)
top10 = rdd.flatMap(lambda item: item.split(' ')) \
    .map(lambda item: re.sub(r'[^a-zA-Z]', '', item).lower()) \
    .filter(lambda item: item != '') \
    .map(lambda item: (item, 1)) \
    .reduceByKey(lambda x, y: x + y) \
    .sortBy(lambda x: x[1], ascending=False) \
    .take(10)
print(top10)

File: Spark/scala/WordCountRDD.scala
import org.apache.spark._
import org.apache.spark.rdd._

object WordCountRDD {
    def main(args : Array[String]) : Unit => {
        val sparkConf = new SparkConf()
            .setMaster("local[*]")
            .setAppName("WordCount_RDD_Scala")
        val sc = SparkContext(sparkConf)

        val filePath = "file://" + args[0]
        val rdd = sc.textFile(filePath)
        val top10 = rdd.flatMap(_.split(' '))
            .map(_.replaceAll("[^a-zA-Z]", "").toLowerCase)
            .filter(_ != "")
            .map((_, 1))
            .reduceByKey(_ + _)
            .sortBy(_._2, ascending=false)
            .take(10)

        top10.foreach(println)
    }
}
